swarm/src/behaviour/external_addresses.rs

```diff
@@ -29,18 +18,40 @@ impl ExternalAddresses {
         self.addresses.iter()
     }
 
+    pub fn as_slice(&self) -> &[Multiaddr] {
+        self.addresses.as_slice()
+    }
+
     /// Feed a [`FromSwarm`] event to this struct.
     ///
     /// Returns whether the event changed our set of external addresses.
     pub fn on_swarm_event<THandler>(&mut self, event: &FromSwarm<THandler>) -> bool {
         match event {
             FromSwarm::ExternalAddrConfirmed(ExternalAddrConfirmed { addr }) => {
-                if self.addresses.len() < self.limit {
-                    return self.addresses.insert((*addr).clone());
+                if self.addresses.contains(addr) {
+                    return false;
```

Comment:
In case the address is already here, shouldn't we move it to the back such that fresh addresses are less likely to be removed? Are `NetworBehaviour`s supposed to continuously report the address?

Reply:
Done.